### PR TITLE
xtimer: refine config macros

### PR DESCRIPTION
**Pull request by smlng**
This PR refines some config macros for xtimer, i.e., XTIMER_SHIFT is auto set relative to XTIMER_HZ.

### CODE REVIEW

**Comment by smlng** [comment]
waiting for #6507

File: sys/include/xtimer.h [hunk 1]
```diff
@@ -573,11 +555,43 @@ int xtimer_mutex_lock_timeout(mutex_t *mutex, uint64_t us);
...
+ *
+ * For a 1 MHz hardware timer, set XTIMER_SHIFT to 0.
+ *
+ * For a 4 MHz hardware timer, set XTIMER_SHIFT to 2.
+ * For a 16 MHz hardware timer, set XTIMER_SHIFT to 4.
+ * For a 250 kHz hardware timer, set XTIMER_SHIFT to 2.
+ *
+ * The direction of the shift is handled by the macros in tick_conversion.h
+ */
+#define XTIMER_SHIFT (0)
```
Comment (kaspar030) [inline sys/include/xtimer.h]: please make this explicit, and err out if the shift values cannot be computed. Otherwise this breaks with e.g., 500kHz.

Comment (smlng) [inline sys/include/xtimer.h]: good point, will adapt. Btw. 500kHz would be `XTIMER_SHIFT (1)` which is covered above, but not mentioned in the docu 😕

**Comment by smlng** [comment]
@kaspar030 may I squash? ... and you ACK 😉

**Comment by smlng** [comment]
can we move on with this?

**Comment by smlng** [comment]
rebased

**Comment by smlng** [comment]
ping @kaspar030

**Comment by smlng** [comment]
@kaspar030 you have a blocking review here, anything to add, otherwise please approve.

**Comment by kYc0o** [comment]
So we forgot #6507 ? 🙄

**Pull request merged**